src/charm.py

```diff
@@ -331,11 +333,14 @@ def set_security_key(self, container):
             :class:`ops.pebble.ConnectionError` if pebble is not ready
         """
         file_path = SECRET_PATH + "/" + KEY_FILE
-        container.push(file_path,
-                       self.security_key,
-                       permissions=0o400,
-                       user="mongodb",
-                       group="mongodb")
+        try:
+            container.push(file_path,
+                           self.security_key,
+                           permissions=0o400,
+                           user="mongodb",
+                           group="mongodb")
+        except (PathError, ProtocolError):
+            self.unit.status = BlockedStatus("Failed to set security key")
```

Comment:
Consider adding logger.error with file_path.

Reply:
logger.error is only obvious if the adminstrator chooses to look at the logs. More than likely the admin will look at juju status. For this reason I would prefer to leave it at BlockedStatus. I can add a logger.error additionally.